### PR TITLE
DCS-409 Starting to work out case sampling

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: src/test/java/uk/gov/justice/hmiprobation/casesampler/utils/SampleSizeTest.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                E to listOfSize(169)
+        ))
+
+        assertThat(sizes).isEqualTo(listOf(
+                SampleSize(A, 55, "37.30"),
+                SampleSize(B, 17, "11.34"),
+                SampleSize(C, 40, "26.87"),
+                SampleSize(D, 16, "10.60"),
+                SampleSize(E, 21, "13.89")
+        ))
```
Comment (andrewrlee) [inline src/test/java/uk/gov/justice/hmiprobation/casesampler/utils/SampleSizeTest.kt:64]: @markberridge  this matches the stats in the sampling doc:
<kbd>Screenshot 2020-05-21 at 10 06 43</kbd>

File: src/main/java/uk/gov/justice/hmiprobation/casesampler/dto/Cases.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    // Ensures, that this case is the earliest of any duplicates in the passed in list
+    fun isEarliestCaseForOffender(cases: List<Case>): Boolean {
+        val earliestCase = cases.asSequence()
+                .filter { isSameOffender(it) }
+                .sortedBy { it.startDate }
+                .first()
+        return this === earliestCase
+    }
+
+    fun getStratification() = when (gender) {
```
Comment (markberridge) [inline src/main/java/uk/gov/justice/hmiprobation/casesampler/dto/Cases.kt]: Should this be "getStratum"? Stratum is the noun you would "get", stratification is the process of partitioning a population into subpopulations according to [wikipedia](https://en.wikipedia.org/wiki/Stratified_sampling)

File: src/main/java/uk/gov/justice/hmiprobation/casesampler/dto/Cases.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+enum class SentenceType {
+    //Community Sentence (CO & SSO)
+    COMMUNITY_SENTENCE,
+
+    // Post-custody (Licence/PSS)
+    POST_CUSTODY
+}
+
+enum class Stratification {
```
Comment (markberridge) [inline src/main/java/uk/gov/justice/hmiprobation/casesampler/dto/Cases.kt]: should this be "Stratum"? see below

File: src/test/java/uk/gov/justice/hmiprobation/casesampler/utils/SampleSizeTest.kt (hunk 1 above)
Comment (markberridge) [reply src/test/java/uk/gov/justice/hmiprobation/casesampler/utils/SampleSizeTest.kt:64]: Cool. I would go with the simplest basic rounding to an integer as you have done until we find out different

**Review by markberridge — CHANGES_REQUESTED** [review]
LGTM other than small language change to use the word "stratum" in place of "stratification"

File: src/main/java/uk/gov/justice/hmiprobation/casesampler/dto/Cases.kt (hunk 2 above)
Comment (andrewrlee) [reply src/main/java/uk/gov/justice/hmiprobation/casesampler/dto/Cases.kt]: Went round the houses with strata, stratum, stratification, type etc.. :laughing: - supposedly stratification and stratum are [synonyms](https://www.thesaurus.com/browse/stratification) but I lost confidence with it. Stratum is better so I'll rename it to that in a future PR

Comment (andrewrlee) [reply src/main/java/uk/gov/justice/hmiprobation/casesampler/dto/Cases.kt]: Renamed it to Stratum

File: src/main/java/uk/gov/justice/hmiprobation/casesampler/dto/Cases.kt (hunk 3 above)
Comment (andrewrlee) [reply src/main/java/uk/gov/justice/hmiprobation/casesampler/dto/Cases.kt]: Changed this

**Pull request merged**